### PR TITLE
[Concurrency] Allow async closures to be annotated with global actors.

### PR DESCRIPTION
**Pull request by DougGregor**
Allow an asynchronous closure to be annotated with a global actor,
which isolates it to that actor. For example:

```swift
Task.runDetached { @mainactor in
  // runs on main actor
}
```

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by DougGregor** [comment]
(This needs more tests)

File: lib/Parse/ParseDecl.cpp [hunk 1]
```diff
@@ -5728,6 +5728,15 @@ void Parser::skipSILUntilSwiftDecl() {
   }
 }
 
+void Parser::skipAnyAttribute() {
+  consumeToken(tok::at_sign);
+  if (!consumeIf(tok::identifier))
+    return;
+
+  if (consumeIf(tok::l_paren))
+    skipUntil(tok::r_paren);
```
Comment (rintaro) [inline lib/Parse/ParseDecl.cpp:5737]: `skipUntil(tok::r_paren)` doesn't consume `)`. I'd suggest

```suggestion
  if (Tok.is(tok::l_paren))
    skipSingle();
```

Comment (rintaro) [reply lib/Parse/ParseDecl.cpp:5737]: On second thought, this is ambiguous `foo { @someAttr (a: Int) -> Int in }`. We should do some heuristics just like: https://github.com/apple/swift/blob/e024d96f59ba26a336b2e88490385ba2a19375d7/lib/Parse/ParseDecl.cpp#L3178-L3186

Comment (DougGregor) [reply lib/Parse/ParseDecl.cpp:5737]: You're right, and if I'd written the tests @slavapestov rightly asked for I would have found this. Thank you! I'll clean this up shortly.

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by DougGregor** [comment]
Unrelated failure in... Foundation? Huh.

**Comment by DougGregor** [comment]
@swift-ci please smoke test macOS

**Comment by DougGregor** [comment]
@swift-ci please smoke test Linux

**Pull request merged**